### PR TITLE
Drop i686

### PR DESCRIPTION
**Pull request by t184256**
Drop i686 building, don't simplify code though.

Draft because I've based it on top of the yet unmerged #150, will rebase after you merge.

### CODE REVIEW

**Comment by Gerschtli** [comment]
Do you plan to simplify the code in a follow-up PR?

**Comment by t184256** [comment]
I'm undecided. These might turn useful again someday, it was just the part where we ship untested zipballs / proot that I didn't like. What do you think?

**Comment by Gerschtli** [comment]
Currently it just adds unnecessary complexity and if necessary we can always look at the PR what we changed. And if we do not test it nor build i686 anymore, we could easily forget updating the stuff for i686, so we have more complexity and if we would introduce it, we would need to test it thoroughly again so or so.

**Comment by t184256** [comment]
OK, I'll do some follow-up cleanup tomorrow.

**Comment by Gerschtli** [comment]
In this or a separate PR?

**Comment by t184256** [comment]
Could be separate.

**Review by Gerschtli — APPROVED** [review]

**Comment by t184256** [comment]
Attempted simplifying the affected code.

File: pkgs/nix-directory.nix [hunk 1]
```diff
@@ -28,7 +28,6 @@ stdenv.mkDerivation {
       let
         archShas = {
          aarch64 = "1hl6pd02nssscn32mrndif2fxfssxiarrpjvqyjicwnz6yn9mhpq";
-         i686 = "0249sbmgmi7cah099wywlqx3ygwpjgl6vcyivqix9rwpnpap417x";
         };
       in
         "${archShas.${config.build.arch}}";
```
Comment (Gerschtli) [inline pkgs/nix-directory.nix:33]: Could also be simplified to
```nix
sha256 = "1hl6pd02nssscn32mrndif2fxfssxiarrpjvqyjicwnz6yn9mhpq";
```

**Comment by t184256** [comment]
sorry for a force-push of the last 4 commits, but I've messed the history up beyond fixupping

File: flake.nix [hunk 2]
```diff
@@ -13,35 +13,31 @@
...
         };
-    }
-    // flake-utils.lib.eachSystem
-      [ "aarch64-linux" ]
-      (system: {
-        apps.nix-on-droid = appPerSystem system;
-        defaultApp = appPerSystem system;
-      });
+        apps.nix-on-droid.aarch64-linux = app;
+        defaultApp.aarch64-linux = app;
```
[the remark this answers is not in the document]
Comment (t184256) [reply flake.nix]: fixed (ab7a4f9)

File: flake.nix [hunk 3]
```diff
@@ -13,35 +13,31 @@
...
+        drv = pkgs'.callPackage ./nix-on-droid { };
       };
     in
     {
       inherit overlay;
 
       lib.nixOnDroidConfiguration =
         { config
-        , system
+        , system ? "aarch64-linux"  # unused
```
[the remark this answers is not in the document]
Comment (t184256) [reply flake.nix]: Like this?

File: pkgs/nix-directory.nix (hunk 1 above)
Comment (t184256) [reply pkgs/nix-directory.nix]: Done (eaf7dec).

File: flake.nix (hunk 3 above)
[the remark this answers is not in the document]
Comment (Gerschtli) [reply flake.nix]: Yes :)

**Review by Gerschtli — APPROVED** [review]
LGTM, thanks!

**Comment by t184256** [comment]
(On one hand, my life's definitely easier when what I work on is a patchset and the UI displays the evolution of that patchset like GitLab, reviewable and probably everything else serious out there do. On the other hand, I don't want to bolt stuff like reviewable onto github, so I'll need to learn, I guess)

File: modules/environment/login/default.nix [hunk 4]
```diff
@@ -86,7 +86,6 @@ in
         let
           crossCompiledPaths = {
            aarch64 = "/nix/store/dapbgzbpl426jrhz4a2sdl096a8l98ad-proot-termux-aarch64-unknown-linux-android-unstable-2021-11-21";
-           i686 = "/nix/store/6nnjhrh8pgyxjnya72irahxpkbnxai1w-proot-termux-i686-unknown-linux-android-unstable-2021-11-21";
           };
         in
           "${crossCompiledPaths.${config.build.arch}}";
```
[the remark this answers is not in the document]
Comment (t184256) [reply modules/environment/login/default.nix]: Done (and build.sh updated)

**Comment by Gerschtli** [comment]
I agree that GitLab handles that in a better way. With GitHub you need to always fix your commits before merge to get a clean history.. Always a bit messy

**Pull request merged**